pkg/util/pod/pod.go

```diff
@@ -76,3 +76,24 @@ func GetResourcePodSpec(fedObject fedcorev1a1.GenericFederatedObject, gvk schema
 	}
 	return GetPodSpec(fedObject, path)
 }
+
+func GetResourcePodSpecFromUnstructuredObj(
```

Comment:
nit: The above func GetPodSpec can reuse this to simplify

Reply:
The common logic of this function has been extracted.